### PR TITLE
Bug 2015793: Run OLM's collect profiles job on the management cluster

### PR DESCRIPTION
**Pull request by awgreene**

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -2081,6 +2073,46 @@ func (r *HostedControlPlaneReconciler) reconcileOperatorLifecycleManager(ctx con
 		return fmt.Errorf("failed to reconcile packageserver worker endpoints: %w", err)
 	}
 
+	// Collect Profiles
+	collectProfilesConfigMap := manifests.CollectProfilesConfigMap(hcp.Namespace)
+	olm.ReconcileCollectProfilesConfigMap(collectProfilesConfigMap, p.OwnerRef, p.OLMImage, hcp.Namespace)
+	if err := r.Create(ctx, collectProfilesConfigMap); err != nil && !apierrors.IsAlreadyExists(err) {
```
Comment (awgreene) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:2079]: The pprof config shouldn't be modified - it allows users to disable the cronjob and the generated configmaps.

Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:2079]: I think we need to decide which user can modify the configmap. If it's the owner of the guest cluster (aka customer), then we should probably create this in the guest cluster and mirror its content back to the control plane. If it's meant to be configured by the cluster provider (aka management), then there likely should be a way to modify it via the HostedCluster API. I wouldn't expect anyone to make changes directly to configmaps in the control plane's namespace.

Comment (awgreene) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:2079]: I believe the owner of the control plane should be responsible for modifying this configMap. It acts as an emergency escape hatch for disabling the cronjob, which runs on the management cluster.

If we want to configure this via the hostedCluster API, we can:
* Modify the hosted control plane operator to create/update this configMap.
* Introduce a knob on the hostedCluster API to set the value in the configMap.

**Comment by awgreene** [comment]
openshift/operator-framework-olm#208 has merged
/unhold

**Pull request merged**